src/tests/ftest/util/storage_utils.py

```diff
@@ -0,0 +1,555 @@
+
+        # Verify each server host has the same NVMe devices behind the same VMD addresses.
+        if not controllers:
+            self._raise_error("Error: Non-homogeneous NVMe device behind VMD addresses.")
+
+        self._log.debug('Controllers found with equal disk quantity on %s', str(self._hosts))
+        for controller, disk_count in controllers.items():
+            self._log.debug('  %s: %s disk(s)', controller, disk_count)
+
+        return controllers
+
+    def write_storage_yaml(self, yaml_file, engines, tier_type, scm_size=100):
+        """Generate a storage test yaml sub-section.
+
+        Args:
+            yaml_file (str): file in which to write the storage yaml entry
+            engines (int): number of engines
+            tier_type (str): storage type to define; 'pmem' or 'md_on_ssd'
+            scm_size (int, optional): scm_size to use with ram storage tiers. Defaults to 100.
+
+        Raises:
+            StorageException: if an invalid storage type was specified
+
+        """
+        tiers = 1
+        self._log.info(
+            'Generating a %s storage yaml for %s engines: %s', tier_type, engines, yaml_file)
+
+        if tier_type not in self.TIER_KEYWORDS:
+            self._raise_error(f'Error: Invalid storage type \'{tier_type}\'')
+
+        if self.devices:
+            # Sort the detected devices and place then in lists by NUMA node
+            numa_devices = {}
+            for device in sorted(self.controller_devices or self.disk_devices):
+                if device.numa_node not in numa_devices:
+                    numa_devices[device.numa_node] = []
+                numa_devices[device.numa_node].append(device.address)
+            self._log.debug('numa_devices:   %s', numa_devices)
+
+            # Interleave the devices for bdev_list distribution.
+            if engines > 1:
+                # This will also even out any uneven NUMA distribution using the shortest list
+                interleaved = list(itertools.chain(*zip(*numa_devices.values())))
+            else:
+                # Include all devices with uneven NUMA distribution
+                interleaved = list(
+                    filter(
+                        partial(is_not, None),
+                        itertools.chain(*itertools.zip_longest(*numa_devices.values()))))
+            self._log.debug('interleaved:    %s', interleaved)
+
+            # Break the interleaved (prioritized) disk list into groups of engine size
+            device_sets = [interleaved[x:x + engines] for x in range(0, len(interleaved), engines)]
+            self._log.debug('device_sets:    %s', device_sets)
+
+            # Tier number device placement order
+            tier_placement_priority = [1]
+
+            # Get the tier number device placement for the available number of devices
+            tier_placement = []
+            while len(tier_placement) < len(device_sets):
+                tier_placement.extend(tier_placement_priority)
```

Comment:
Just fun theoretical stuff:
```suggestion
            tier_placement = math.ceil(len(device_sets) / len(tier_placement)) * tier_placement_priority
```